### PR TITLE
[IE Samples] Enable a custom build output folder

### PR DESCRIPTION
**Pull request by dpigasin**

### CODE REVIEW

File: inference-engine/samples/build_samples.sh [hunk 1]
```diff
@@ -3,6 +3,44 @@
...
+    echo "Options:"
+    echo "  -h                       Print the help message"
+    echo "  -b SAMPLE_BUILD_DIR      Specify the sample build directory"
+    echo "  -i SAMPLE_INSTALL_DIR    Specify the sample install directory"
+    echo
+    exit 1
+}
+
+samples_type=$(basename "$PWD")
+build_dir="$HOME/inference_engine_${samples_type}_samples_build"
```
[the remark this answers is not in the document]
Comment (dpigasin) [reply inference-engine/samples/build_samples.sh:18]: What should $samples_type be equal to? I guess it is parsing incorrectly.

File: inference-engine/samples/CMakeLists.txt [hunk 2]
```diff
@@ -245,3 +245,10 @@ if(EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/thirdparty")
     list(REMOVE_ITEM samples_dirs common thirdparty)
 endif()
 add_samples_to_build(${samples_dirs})
+
+install(
+    DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/intel64/Release/
```
[the remark this answers is not in the document]
Comment (ilya-lavrenov) [reply inference-engine/samples/CMakeLists.txt]: Why do we need to debug samples?

File: inference-engine/samples/CMakeLists.txt [hunk 3]
```diff
@@ -245,3 +245,10 @@ if(EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/thirdparty")
     list(REMOVE_ITEM samples_dirs common thirdparty)
 endif()
 add_samples_to_build(${samples_dirs})
+
+install(
+    DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/${ARCH}/${CMAKE_BUILD_TYPE}/
```
Comment (ilya-lavrenov) [inline inference-engine/samples/CMakeLists.txt:250]: let's use `install(TARGET ..)`

File: inference-engine/samples/common/format_reader/CMakeLists.txt [hunk 4]
```diff
@@ -40,3 +40,10 @@ set_target_properties(${TARGET_NAME} PROPERTIES COMPILE_PDB_NAME ${TARGET_NAME}
 if(COMMAND add_clang_format_target)
     add_clang_format_target(${TARGET_NAME}_clang FOR_TARGETS ${TARGET_NAME})
 endif()
+
+install(
+        TARGETS ${TARGET_NAME}
+        RUNTIME DESTINATION samples_bin/
```
Comment (ilya-lavrenov) [inline inference-engine/samples/common/format_reader/CMakeLists.txt:46]: why `LIBRARY DESTINATION` is not used here as well? Here it's https://github.com/openvinotoolkit/openvino/blob/master/samples/CMakeLists.txt#L18-L19

The same for OpenCV C wrapper

**Review by antonrom23 — APPROVED** [review]

**Review by ilya-lavrenov — APPROVED** [review]

**Pull request merged**